src/pointer-analysis/value_set.cpp

```diff
@@ -1288,7 +1288,7 @@ void value_sett::assign_rec(
   else if(
     is_constant_string2t(lhs) || is_null_object2t(lhs) ||
     is_valid_object2t(lhs) || is_deallocated_obj2t(lhs) ||
-    is_dynamic_size2t(lhs) || is_constant_array2t(lhs))
+    is_dynamic_size2t(lhs) || is_constant_array2t(lhs) || is_invalid2t(lhs))
```

Comment:
This check should be done where the invalid2t expression is obtained originally in the recursion: in the `is_dereference2t(lhs)` branch. The reason is that when possible we should handle the "special" value-set expressions `unknown2t`, `invalid2t`, etc. together, such that it's clear that `assign_rec()` itself only handles the non-special ones; i.e., the special should not escape.

Reply:
I see, thank you!